code-scanning/fortify-on-demand.yml

```diff
@@ -0,0 +1,97 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+################################################################################################################################################
+# Fortify lets you build secure software fast with an appsec platform that automates testing throughout the DevSecOps pipeline. Fortify static,#
+# dynamic, interactive, and runtime security testing is available on premises or as a service. To learn more about Fortify, start a free trial #
+# or contact our sales team, visit microfocus.com/appsecurity.                                                                                 #
+#                                                                                                                                              #
+# Use this workflow template as a basis for integrating Fortify on Demand Static Application Security Testing(SAST) into your GitHub workflows.#
+# This template demonstrates the steps to prepare the code+dependencies, initiate a scan, download results once complete and import into       #
+# GitHub Security Code Scanning Alerts. Existing customers should review inputs and environment variables below to configure scanning against  #
+# an existing application in your Fortify on Demand tenant. Additional information is available in the comments throughout the workflow, the   #
+# documentation for the Fortify actions used, and the Fortify on Demand / ScanCentral Client product documentation. If you need additional     #
+# assistance with configuration, feel free to create a help ticket in the Fortify on Demand portal.                                            #
+################################################################################################################################################
+
+name: Fortify on Demand Scan
+
+# TODO: Customize trigger events based on your DevSecOps processes and typical FoD SAST scan time
+on:
+  workflow_dispatch:
+  push:
```

Comment:
Was it a deliberate decision to deviate from the standard code scanning triggers? In order to show PR analyses the workflow will also need to trigger on pull request events.

Reply:
to clarify, the standard triggers are as seen in the CodeQL workflow...

```
  push:
    branches: [ $default-branch, $protected-branches ]
  pull_request:
    # The branches below must be a subset of the branches above
    branches: [ $default-branch ]
  schedule:
    - cron: $cron-weekly
```

Reply:
Files renamed as requested. The pull_request trigger was left out deliberately; although it will work, there are some subtle considerations to be taken into account to properly handle scan results for multiple branches or pull requests on the server side. So while we are working out the details, we've chosen to not actively promote the pull_request trigger.

Reply:
Thanks for the naming update, and for the details! Could you expand a bit more on how Fortify handles analysing different commits? On our side code scanning should be fine so long as each delivery contains all the alerts that are present in a given commit (eg head of the default branch or merge commit of a PR). Does your tool also do some sort of diffing...?

Reply:
On the GitHub side everything should be fine; this workflow will report the appropriate issues for the code that was checked out. The problem is that on our server side, by default users just see the results of the latest scan without really knowing whether that was a scan of the main branch or a PR. We are working out a process that will improve this, but in the meantime we need to replace the existing workflow to use non-deprecated actions.